plugins/win-wasapi/win-wasapi.cpp

```diff
@@ -231,7 +229,67 @@ bool WASAPISource::InitDevice(IMMDeviceEnumerator *enumerator)
 		bfree(w_id);
 	}
 
-	return SUCCEEDED(res);
+	return res;
+}
+
+HRESULT WASAPISource::InitDeviceLoop(IMMDeviceEnumerator *enumerator,
+				     int maxRetry, int sleepMs)
+{
+	HRESULT res = -1;
+	int retryInitDeviceCounter = 0;
+	std::vector<AudioDeviceInfo> devices;
+
+	while (retryInitDeviceCounter < maxRetry) {
+		//if for first time, use the isDefaultDevice flag, otherwise,
+		// use the current device_id which could have been changed for the same device name
+		res = InitDevice(enumerator, !retryInitDeviceCounter
+						     ? isDefaultDevice
+						     : false);
+		
+		if (device_name.empty())
+			device_name = GetDeviceName(device);
+		
+		/*
+		`SUCCEEDED(res)` needs to be after getting the device name (if it's empty). 
+		Then , if res has not succeeded and  the device name is not empty
+		i.e. it took the device name successfully at least once,
+		search in the audio device list for that device name.
+		If res has succeeded, exit early. */
+
+
+		if (SUCCEEDED(res)) {
+			break;
+		}
+		if (!device_name.empty()) {
+			blog(LOG_INFO,
```

Comment:
if the device name is not empty and res has not succeeded, get the new list of audio devices and loop to get a device id with the same device name